### ISSUE TITLE
4k/high-DPI support

### ISSUE THREAD

**Issue by andyjones789** [digitalcreations/MaxTo#30]
I have a 4k display and when I use the Change Regions feature, the box doesn't fill the screen and says the screen size is 2560x1413. 

What's wrong?

**Comment by vegardlarsen** [comment]
@andyjones789 I am painfully aware of this issue, but I haven't posted about it publicly as of yet. At the moment, MaxTo offers no support for high-DPI screens. The issue is harder than it appears at first because of a few issues (my grasp of which is truthfully not complete just yet):

Starting with Windows 8, you can have different DPI settings for different monitors. You can see this clearly when you drag a window from a standard DPI-monitor to a high-DPI monitor. Windows then handles the upscaling/downscaling for you. 

Now, most programs running on Windows today are not DPI-aware, meaning that Windows simply scales the windows when you are running on high-DPI. This gives the impression of pixelation. Sometimes, programs claim to support high DPI but does so badly.

MaxTo's problem here is that parts of MaxTo actually runs inside multiple other programs, which means it needs to know if each individual program is DPI-aware or not.

This is where the details are a bit murky for me yet, as I haven't had the time to figure this out completely yet. When MaxTo sends instructions to a window to resize itself, MaxTo may (or may not, depending on the situation) need to translate the coordinates it uses to/from DPI-aware coordinates. While the calculation itself is easy, knowing when to perform it is not trivial.

**TL;DR**: disable high-DPI for now to work with MaxTo. A fix will be released, but it might take a while.

**Comment by andyjones789** [comment]
Hi,

Thank you for your response - we are software developers ourselves (but
thankfully only work in web technology so are mostly spared these sorts of
issues).

I wish you luck in fixing this issue and I'd be more than happy to
alpha/beta test for you.

Thanks again,
Andy

**Comment by MLeVell** [comment]
I have a new 4k Dell P2815Q Monitor and was looking for a multi-monitor solution.  Was just about to buy, but wanted to check the issue list first.  This stopped me... ping me when it's fixed!  Thanks.  -Mark

**Comment by nickfie** [comment]
I've had no problem using MaxTo with Seiki 1 Meter/39 inch UHD (3840x2160) acting as monitor, in addition to laptop screen. Provides best value from the enormous screen area.

Normal measures should be taken. Ensure everything in the chain is most recent:
- System BIOS
- OS patches
- PC vendor drivers and Video card driver(s). My Lenovo laptop has both Intel integrated video and NVidia card, so this is complicated. 
... Apply Lenovo version of Intel video driver.
... Download most current NVidia driver from their site and install. Reboot. This is best UHD support.
... Apply Lenovo's NVidia driver from their site. This keeps newer NVidia code and overlays integration with Intel drivers & Lenovo BIOS.
- DisplayPort 1.2 cables from reputable vendor. UHD is very high bandwidth.
- If you must convert to HDMI, look for "Active" "2-way" adapters with lots of good reviews. Bandwidth.
- Short, high-bandwidth HDMI cables. Check for many good reviews from UHD users.

Seiki unit was inexpensive. I had to get every item listed above correct before I had a rock-solid reliable monitor. Adding MaxTo has raised my productivity tremendously.

**Comment by raulajr** [comment]
Interesting observation: I was experiencing the same issue with my Windows 10. I have a laptop (15" screen) and a monitor (23") screen with the screen extended. I had the issue on the laptop screen but the monitor was fine. When disabled "Use small taskbar buttons" the issue went away. I re-enable it and it comes back. One other thing I noticed is that when I have the issue, the taskbar notification tray icons appear pixelated. When I change the setting, icons are normal. 

Hopefully this helps others out there until MaxTo permanently fixes the issue.

**Comment by YammYcoding** [comment]
![maxto-bug-hdmi](https://cloud.githubusercontent.com/assets/1200923/25524884/aa87eb7c-2be2-11e7-9566-5269fdb272a6.jpg)

This happened to me when I removed the dvi cable from the display and plugged an hdmi cable. Nvidia has also updated driver on the same day, and the maxtoversion wasn't changed, so It could be the nvidia driver or the hdmi cable, or even both.

Now I can't use maxto anymore. please fix it :(

**Comment by vegardlarsen** [comment]
`<voice person="Professor Hubert J. Farnsworth">`Good news, everyone!`</voice>`

Microsoft announced .NET Framework 4.7 General Availability on Tuesday, and it contains a change to Windows Forms which lets MaxTo work more nicely with mixed DPI screens. To use this, we had to change a few settings and recompile MaxTo for .NET Framework 4.7, and it should provide a better experience on all platforms, but it is not perfect.

Depending on which version of Windows you have, the control may either be very large on normal screens or very small on high-DPI screens. I believe this may also vary based on which monitor takes precedence when starting Windows.

The good news is that this fix removes the "quarter-screen" issue completely. To make this work on your machine, you need to [download and install .NET Framework 4.7](https://www.microsoft.com/en-us/download/details.aspx?id=55170).

Then download and unzip these files over the existing files in `C:\Program files\MaxTo`: [Fix.zip](https://github.com/digitalcreations/MaxTo/files/978518/Fix.zip)

Restart MaxTo, and it should be per-monitor DPI aware.

Please let me know of any issues you may experience with this fix. Thank you for waiting for us! :)

PS! This will make it into a proper release shortly, but the installer software does not support .NET Framework 4.7 yet.
PPS! This does not support Windows 7 and 8 (but it does support 8.1). MaxTo's requirements will change when this is released.

**Comment by jdunham22** [comment]
It works!  awesome, thanks.

**Comment by xckai** [comment]
win10 enterprise has no .net 4.7  0.0

**Comment by vegardlarsen** [comment]
@xckai It does. It is installed with the creator's update. MaxTo was just released earlier today with this fix baked in. So simply go download version 2017.07 from the website and it should work.

**Comment by raulajr** [comment]
@vegardlarsen My understanding is that Windows 10 users need the Anniversary Update to receive 4.7 but I cannot upgrade from Win10 Enterprise to the Anniversary version because my workstation is locked down by my company. Is there another way?
